src/smt/solver_engine.cpp

```diff
@@ -2148,6 +2148,10 @@ void SolverEngine::setOption(const std::string& key,
 {
   if (fromUser && options().base.safeOptions)
   {
+    if (key == "trace")
+    {
+      throw OptionException("cannot use trace messages with safe-options");
+    }
```

Comment:
why this limitation?

Reply:
Line 2156 below fails for `-t`, since it is not a normal option, so this is to guard against this.

Moreover, in general trace messages should not be used in production.